src/rgw/rgw_auth.cc

```diff
@@ -445,6 +488,23 @@ void rgw::auth::WebIdentityApplier::modify_request_state(const DoutPrefixProvide
   condition.clear();
   condition = idp_url + ":sub";
   s->env.emplace(condition, token_claims.sub);
+
+  if (principal_tags) {
+    std::unordered_map<string, string> tags_map;
+    string_to_map(dpp, tags_map);
+    for (auto& it : tags_map) {
```

Comment:
If we're just going to make a map, iterate over it, then throw the map away, that's a pattern I'd rather avoid. Consider having a function template that takes a function and calls it on each KV pair in the string.

Or make an iterator similar to Casey's spliterator (common/split.h)

Reply:
I like the function template accumulator idea, esp if its kind of a one-off

Reply:
It probably could. The main difference is that this splits the key from the value and spliterator you'd have to do that yourself. It'd probably be just as easy to do that and make two string views.